pkg/webservice/handlers.go

```diff
@@ -339,3 +341,63 @@ func getClusterConfig(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 }
+
+func updateConfig(w http.ResponseWriter, r *http.Request) {
+	lock.Lock()
+	defer lock.Unlock()
+	writeHeaders(w)
+	requestBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	// validation is already called when loading the config
+	schedulerConf, err := configs.LoadSchedulerConfigFromByteArray(requestBytes)
+	if err != nil {
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	oldConf, err := updateConfiguration(string(requestBytes))
+	if err != nil {
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	err = gClusterInfo.UpdateSchedulerConfig(schedulerConf)
+	if err != nil {
+		// revert configmap changes
+		_, err := updateConfiguration(oldConf)
+		if err != nil {
+			log.Logger().Error("Configuration rollback failed")
+		}
+		buildUpdateResponse(false, err.Error(), w)
+		return
+	}
+	buildUpdateResponse(true, "", w)
+}
+
+func buildUpdateResponse(success bool, reason string, w http.ResponseWriter) {
```

Comment:
for the REST API response. If update was successful, we should do

```
json.NewEncoder(w).Encode(result)
```

however, if the update was not successful, I think we should return an error code to the client

```
http.Error(w, err.Error(), http.StatusConflict))
```

where the StatusConflict means:  The request could not be completed due to a conflict with the current state of the target resource. This code is used in situations where the user might be able to resolve the conflict and resubmit the request.

Reply:
Changed it

Reply:
@yangwwei , If we return StatusConflict in case of failure, I think there is no need to have an UpdateConfResponse, because we can set the status code to 200-OK and return a message that the config update was successful, so I removed the UpdateConfResponse and set 200-OK in case of success.